pallene/types.lua

```diff
@@ -98,6 +138,24 @@ local function equivalent(t1, t2, is_gradual)
     elseif tag1 == "types.T.Array" then
         return equivalent(t1.elem, t2.elem, is_gradual)
 
+    elseif tag1 == "types.T.Table" then
+        local f1 = t1.fields
+        local f2 = t2.fields
+
+        for name, typ in pairs(f1) do
+            if not f2[name] or not equivalent(typ, f2[name], is_gradual) then
+                return false
+            end
+        end
+
+        for name, typ in pairs(f2) do
+            if not f1[name] or not equivalent(f1[name], typ, is_gradual) then
+                return false
+            end
+        end
+
```

Comment:
What about rewriting this using 3 loops? This way we would only call `equivalent` once instead of twice.
1. Check that all fields in t1 are present in t2
2. Check that all fields in t2 are present in t1
3. Check that all fields in t1 are equivalent to the corresponding fields in t2